src/mpid/ch4/src/ch4_progress.h

```diff
@@ -137,7 +137,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_progress_test(MPID_Progress_state * state, in
 #else
     /* multiple vci */
     if (MPIDI_do_global_progress()) {
-        for (int vci = 0; vci < MPIDI_global.n_total_vcis; vci++) {
+        for (int vci = 0; vci < MPIDI_global.n_vcis; vci++) {
```

Comment:
This is good. But as we discussed, we should skip checking global progress altogether when it is reserved vci, something like:
```
bool is_reserved_vci = (state->vci_count == 1 && state->vcis[0] >= MPIDI_global.n_vcis);
if (!is_reserved_vci && MPIDI_do_global_progress()) {
   ...
```

Reply:
Added as suggested.